Automattic/woocommerce-payments
# Fail upload when dispute evidence goes over limit

Fixes #559 

#### Changes proposed in this Pull Request

- keep track of the file sizes of the dispute evidence
- fail the upload that puts the total size over limit and show an error message to the merchant 

#### Testing instructions

- Check out with a dispute test card (e.g. 4000000000000259)
- Navigate to the Payments » Disputes » Dispute Details » Challenge Dispute screen for the new dispute
- Upload files one after the other.
- Upload a file that puts the combined size over 4.5 MB
- Verify that the file upload fails and failure reason is displayed

<img width="706" alt="Screenshot 2021-06-04 at 6 27 07 PM" src="https://user-images.githubusercontent.com/32092402/120806472-85f8ab00-c564-11eb-8106-3392a3cc90d0.png">

**Notes**
- The check fails if you upload files in parallel which puts the combined size over 4.5 MB. In this case, the form will be submitted but the Stripe error fails the upload.
- Dispute evidence limit on Stripe is 5 MB according to thread p1586212168010800-slack-C9976E5MJ, but on practical testing, it seems to be 4.5 MB. To avoid a scenario where this check allows the upload, but Stripe fails it on submission, 4.5 MB is implemented.
- Unit tests could not be implemented for this code as the doUpload method could not be tested.

-------------------
